controllers/oauth.go

```diff
@@ -0,0 +1,118 @@
+// Handler function for routes
+package controllers
+
+import (
+	"encoding/json"
+	"kwoc-backend/utils"
+	"net/http"
+
+	"kwoc-backend/models"
+)
+
+type OAuthReqBodyFields struct {
+	// Code generated by Github OAuth
+	Code string `json:"code"`
+	// `mentor` or `student`
+	Type string `json:"type"`
+}
+
+type OAuthResBodyFields struct {
+	Username string `json:"username"`
+	Name     string `json:"name"`
+	Email    string `json:"email"`
+	// `mentor` or `student`
+	Type string `json:"type"`
+	// Whether the user has newly registered or was registered before
+	IsNewUser bool   `json:"isNewUser"`
+	Jwt       string `json:"jwt"`
+}
+
+func (dbHandler *DBHandler) OAuth(w http.ResponseWriter, r *http.Request) {
+	db := dbHandler.db
+
+	var reqFields = OAuthReqBodyFields{}
+	err := json.NewDecoder(r.Body).Decode(&reqFields)
+
+	if err != nil {
+		utils.LogErrAndRespond(r, w, err, "Error parsing JSON body parameters.", http.StatusBadRequest)
+		return
+	}
+
+	if reqFields.Code == "" || reqFields.Type == "" {
+		utils.LogWarnAndRespond(r, w, "Empty body parameters.", http.StatusBadRequest)
+		return
+	}
+
+	// Get a Github OAuth access token
+	accessToken, err := utils.GetOauthAccessToken(reqFields.Code)
+	if err != nil {
+		utils.LogErrAndRespond(r, w, err, "Error getting OAuth access token.", http.StatusInternalServerError)
+		return
+	}
+
+	// Get the user's information from the Github API
+	userInfo, err := utils.GetOauthUserInfo(accessToken)
+	if err != nil {
+		utils.LogErrAndRespond(r, w, err, "Error getting OAuth user info.", http.StatusInternalServerError)
+		return
+	}
+
+	if userInfo.Username == "" {
+		utils.LogWarnAndRespond(r, w, "Could not get username from the Github API.", http.StatusInternalServerError)
+		return
+	}
+
+	// Check if the user has already registered
+	var isNewUser bool = false
+
+	if reqFields.Type == "student" {
```

Comment:
Can you create variable for the type student and mentor and use the same when inserting in the database, so that change needs to be done once. Allows better checking for type of the user

Reply:
I didn't fully understand what you mean. Could you elaborate?

Reply:
Just meant that having such keywords used as strings in multiple places is not a good way. If we need to change `student` is say `stud` , we need to find all iterations where this needs to be changes and might even break things where this student was not inferred. Instead Having variables (other languages have enum for this, go doesnt) helps in this case.

Reply:
Resolved.